File: SubscriptionApi/SubscriptionApi/ResponseBuilders/SubscriptionStopResponseBuilder.cs
using System;
using SubscriptionApi.Dto;

namespace SubscriptionApi.ResponseBuilders
{
  public interface SubscriptionStopResponseBuilder : 
    SubscriptionValidationResults, 
    SubscriptionStopEvents,
    UserAuthorizationEvents
  {
    StopSubscriptionResponseDto Build();
    void AssertNoFatalErrors(Exception exception);
  }
}

File: Java/factories/_01_no_factories/src/main/java/ResponseBuilders/SubscriptionStopResponseBuilder.java
package responseBuilders;

import dto.StopSubscriptionResponseDto;

public interface SubscriptionStopResponseBuilder extends
    SubscriptionValidationResults,
    SubscriptionStopEvents,
    UserAuthorizationEvents
  {
    StopSubscriptionResponseDto buildStop();
    void assertNoFatalErrors(RuntimeException exception);
  }
